### ISSUE TITLE
Redundant else if and condition check in `nvme_process_sq_io`

### ISSUE THREAD

**Issue by Sober7135** [MoatLab/FEMU#152]
1. The 1 && in the if condition is unnecessary
2. The else if (status == NVME_SUCCESS) block will never be reached. 

https://github.com/MoatLab/FEMU/blob/1c2d358c1c6bc12ea38c17fb834a4470a53722e5/hw/femu/nvme-io.c#L76-L85

**Comment by huaicheng** [comment]
Yes, you're right. The code was there for debugging purposes, long time ago.

**Comment by huaicheng** [comment]
Confirmed fixed on master — thanks for the report, and sorry it sat open so long.

Both points you raised are gone: the `1 &&` and the unreachable `else if (status == NVME_SUCCESS)` were removed in d749928e9 ("Implement TRIM Support for Dataset Management Command in BBSSD", #174, by @owunr). `nvme_process_sq_io()` now sets `req->status` once and enqueues unconditionally, so there is no second success branch to be unreachable.

Closing.
